modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteInMemoryNodeRestartTest.java

```diff
@@ -195,31 +200,29 @@ public void inMemoryNodeRestartNotLeader(TestInfo testInfo) throws Exception {
         InternalTableImpl internalTable = (InternalTableImpl) restartingTable.internalTable();
 
         // Check that it restarts.
-        assertTrue(waitForCondition(
-                () -> {
-                    boolean raftNodeStarted = loza.localNodes().stream().anyMatch(nodeId -> {
-                        if (nodeId.groupId() instanceof TablePartitionId) {
-                            return ((TablePartitionId) nodeId.groupId()).tableId() == table.tableId();
-                        }
+        waitForCondition(
+                () -> isRaftNodeStarted(table, loza) && assignmentsContain(restartingNodeConsistentId, internalTable),
+                TimeUnit.SECONDS.toMillis(10)
+        );
 
-                        return false;
-                    });
+        assertTrue(isRaftNodeStarted(table, loza), "Raft node is not started");
```

Comment:
Witch node?

Reply:
Raft node corresponding to the partition of interest (that was waited for by the waiter above)

Reply:
Can you add in the error message which node are talking about?